### ISSUE TITLE
Ensure lifetime of BuildCheckManager

### ISSUE THREAD

**Issue by JanKrivanek** [dotnet/msbuild#10317]
### Context

`BuildCheckManager` is currently created once per node and survives for the entire lifetime of the node. This might be problematic as it contains state that should not be shared between builds - BuildChecks registration, ConfigurationModule, etc.

### Possible solution

`BuildRequestEngine` contains `InitializeForBuild` and `CleanupForBuild` methods that anchor a single build. `CleanupForBuild` is already used to finalize the BuildCheckManager stats collection. We can add caches invalidation (if needed in the future - not now).

The `InitializeForBuild` should probably contain code ensuring that `BuildCheckManager` is recreated or reinitialized.

### Alternative solution

https://github.com/dotnet/msbuild/pull/10009 will eventually contain solution for making `BuildCheckManager` instance based and probably owned by `LoggingContext` - so lifetime might not be of a concern anymore... but we are not there yet

**BLOCKED BY** https://github.com/dotnet/msbuild/issues/10145

**Comment by AR-May** [comment]
This issue affects `PropertiesUsageCheck` behavior, as it is created on the out of process node. As node reuse is the default, if you run some solution build without /check and then run build with /check, `PropertiesUsageCheck` will not provide any results for the projects built in the reused out of process nodes.

**Comment by JanKrivanek** [comment]
Note: MSBuild server migh need special handling - as it doesn't dispose singleton components:

https://github.com/dotnet/msbuild/blob/db5b3efcd3c183bd18266c6e261360df91d1aede/src/MSBuild/XMake.cs#L1690-L1693
